builder-base/scripts/install_golang.sh

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -o pipefail
+set -x
+
+SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd -P)"
+
+# go steps are not updated to support /newroot concept yet
+USR=/usr
+USR_LOCAL=/usr/local
+
+source $SCRIPT_ROOT/common_vars.sh
+
+VERSION="$1"
+
+# default, installed first
+GOLANG_VERSION="${GOLANG_VERSION:-1.18.6}"
+GOLANG_DOWNLOAD_URL="https://go.dev/dl/go$GOLANG_VERSION.linux-$TARGETARCH.tar.gz"
+
+
+function build::go::symlink() {
+    local -r version=$1
+
+    # Removing the last number as we only care about the major version of golang
+    local -r majorversion=${version%.*}
+    mkdir -p ${GOPATH}/go${majorversion}/bin
+    for binary in go gofmt; do
+        ln -s /root/sdk/go${version}/bin/${binary} ${GOPATH}/go${majorversion}/bin/${binary}
+    done
+    ln -s ${GOPATH}/bin/go${version} ${GOPATH}/bin/go${majorversion}
+}
+
+function build::go::install(){
+    # Set up specific go version by using go get, additional versions apart from default can be installed by calling
+    # the function again with the specific parameter.
+    local version=$1
+
+    # AL2 provides a longer supported version of golang, use AL2 package when possible
+    local yum_provided_versions="1.13"
+    local eks_built_versions="1.16.15 1.15.15"
+    if [[ $eks_built_versions =~ (^|[[:space:]])${version}($|[[:space:]]) && $TARGETARCH == "amd64" && $IS_AL22 == false ]]; then
+        local artifacts_bucket='eks-d-postsubmit-artifacts'
+        local arch='x86_64'
+        for artifact in golang golang-bin golang-race; do
+          curl https://$artifacts_bucket.s3.amazonaws.com/golang/go/go$version/RPMS/$arch/$artifact-$version-1.amzn2.0.1.$arch.rpm -o /tmp/$artifact-$version-1.amzn2.0.1.$arch.rpm
```

Comment:
we can update this to mirror: https://github.com/aws/eks-distro-build-tooling/pull/534/files, which has yet to be merged. This will point it to the 'prod'  artifacts for EKS Go